### PR TITLE
chore(derived_code_mappings): Ignore another error

### PR DESCRIPTION
**Pull request by armenzg**
Processing of repositories will continue even if we can't reach Github's host.

Fixes [SENTRY-YQ2](https://sentry.sentry.io/issues/3920518293).

### CODE REVIEW

File: src/sentry/integrations/github/client.py [hunk 1]
```diff
@@ -219,6 +219,8 @@ def process_error(error: ApiError, extra: Dict[str, str]) -> None:
                 logger.warning("Github has blocked this org. We will not continue.", extra=extra)
                 # Raising the error will be handled at the task level
                 raise error
+            elif txt.startswith("Unable to reach host:"):
+                logger.warning("Unable to reach host at the moment. Continue.", extra=extra)
```
Comment (snigdhas) [inline src/sentry/integrations/github/client.py:223]: Nit: None of the other errors mention `continue` when they continue - we should stay consistent with one of the two approaches.

File: src/sentry/integrations/github/client.py [hunk 2]
```diff
@@ -215,6 +215,12 @@ def process_error(error: ApiError, extra: Dict[str, str]) -> None:
                 logger.warning(f"The app does not have access to the repo. {msg}", extra=extra)
             elif txt == "Repository access blocked":
                 logger.warning(f"Github has blocked the repository. {msg}", extra=extra)
+            elif txt == "Server Error":
```
Comment (armenzg) [inline src/sentry/integrations/github/client.py:218]: Fixes [SENTRY-YRS](https://sentry.sentry.io/issues/3922127853)

File: src/sentry/integrations/github/client.py [hunk 3]
```diff
@@ -215,6 +215,12 @@ def process_error(error: ApiError, extra: Dict[str, str]) -> None:
...
+                logger.warning(f"Github failed to respond. {msg}.", extra=extra)
+            elif txt == "Bad credentials":
```
Comment (armenzg) [inline src/sentry/integrations/github/client.py:220]: Fixes [SENTRY-YKF](https://sentry.sentry.io/issues/3915010727)

File: src/sentry/integrations/github/client.py [hunk 4]
```diff
@@ -215,6 +215,12 @@ def process_error(error: ApiError, extra: Dict[str, str]) -> None:
...
+                logger.warning(f"No permission granted for this repo. {msg}.", extra=extra)
+            elif txt.startswith("Unable to reach host:"):
```
Comment (armenzg) [inline src/sentry/integrations/github/client.py:222]: Fixes [SENTRY-YQ2](https://sentry.sentry.io/issues/3920518293).

**Review by snigdhas — APPROVED** [review]

**Pull request merged**